### PR TITLE
Update Lucene to 9.8.0

### PR DESCRIPTION
**Pull request by mkleen**
https://lucene.apache.org/core/9_8_0/changes/Changes.html

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java [hunk 1]
```diff
@@ -68,4 +75,25 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
...
+                return format.fieldsWriter(state);
+            }
+
+            @Override
+            public KnnVectorsReader fieldsReader(SegmentReadState state) throws IOException {
+                return format.fieldsReader(state);
+            }
+
+            @Override
+            public int getMaxDimensions(String fieldName) {
```
Comment (mkleen) [inline server/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java:94]: See https://github.com/apache/lucene/pull/12436 for details.

**Review by mfussenegger — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java [hunk 2]
```diff
@@ -68,4 +75,25 @@ public DocValuesFormat getDocValuesFormatForField(String field) {
...
+                return 2048;
```
Comment (matriv) [inline server/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java:95]: Can we please extract this to a constant to have it more visible?

**Review by matriv — APPROVED** [review]
nice! I left a personal pref comment to use a constant for the limit.

File: server/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java (hunk 2 above)
Comment (mkleen) [reply server/src/main/java/org/elasticsearch/index/codec/PerFieldMappingPostingFormatCodec.java]: 👍

File: server/src/main/java/io/crate/types/FloatVectorType.java [hunk 3]
```diff
@@ -53,6 +53,7 @@ public class FloatVectorType extends DataType<float[]> implements Streamer<float
     public static final String NAME = "float_vector";
     public static final FloatVectorType INSTANCE_ONE = new FloatVectorType(1);
     public static final VectorSimilarityFunction SIMILARITY_FUNC = VectorSimilarityFunction.EUCLIDEAN;
+    public static int MAX_DIMENSIONS = 2048;
```
Comment (matriv) [inline server/src/main/java/io/crate/types/FloatVectorType.java:56]:
```suggestion
    public static final int MAX_DIMENSIONS = 2048;
```
or there was a reason you left out `final`?

Comment (mkleen) [reply server/src/main/java/io/crate/types/FloatVectorType.java:56]: 👍

**Pull request merged**